app/helpers/samples_helper.rb

```diff
@@ -122,8 +122,30 @@ def samples_pipeline_run_info(samples)
     pipeline_run_info
   end
 
-  def samples_info(samples)
-    { final_result: samples_output_data(samples),
-      pipeline_run_info: samples_pipeline_run_info(samples) }
+  def filter_samples(samples, query)
+    filtered = []
+    samples.each do |output|
+      pipeline_run_status = output.pipeline_runs.first ? output.pipeline_runs.first.job_status : nil
+      if query == 'UPLOADING'
+        filtered.push(output) unless pipeline_run_status
+      elsif pipeline_run_status == query
+        filtered.push(output)
+      end
+    end
+    filtered
+  end
+
+  def format_samples(samples)
+    formatted_samples = []
+    samples.each_with_index do |_sample, i|
+      job_info = {}
+      final_result = samples_output_data(samples)
+      pipeline_run_info = samples_pipeline_run_info(samples)
+      job_info[:job] = samples[i]
+      job_info[:statistics] = final_result[i]
```

Comment:
this is also confusing, because there's job_stats,  let's call  it something  like derived_sample_output